custom_components/midea_ac_lan/midea/devices/c3/message.py

```diff
@@ -168,7 +168,7 @@ def __init__(self, body, data_offset=0):
             + (body[data_offset + 7] << 8)
             + (body[data_offset + 8])
         )
-        self.outdoor_temperature = int(body[data_offset + 9])
+        self.outdoor_temperature = body[data_offset + 9] - 256 if body[data_offset + 9] > 127 else body[data_offset + 9]
```

Comment:
```suggestion
        base_value = body[data_offset + 9]
        self.outdoor_temperature = (base_value - 256) if base_value > 127 else base_value
```

Reply:
I tested the change with my own heat pump and it works.